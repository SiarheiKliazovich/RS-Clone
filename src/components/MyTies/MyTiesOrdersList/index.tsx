import { FunctionComponent } from "react";

import { IOrdersListComponent } from "../../../interfaces/ordersListComponent";
import { useTranslation } from "react-i18next";

const MyTiesOrdersList: FunctionComponent<IOrdersListComponent> = ({
  orders,
}) => {
  const { t } = useTranslation("dataLang");

  return (
    <div className="market-block__products">
      <div className="row__products">
        {orders.map((order, i) => {
          return (
            <div key={i} className="tie__product">
              <div className="tie__products_img">
                <img src={order.image} alt="" className="products_img" />
              </div>
              <div className="tie__products_discription">
                <div className="tie__products_name">
                  {t("myTiesOrders.date")}: {order.date.slice(0, 10)}
                </div>
                <div className="tie__products_price">
                  {t("myTiesOrders.sum")}: {order.price}$
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default MyTiesOrdersList;
