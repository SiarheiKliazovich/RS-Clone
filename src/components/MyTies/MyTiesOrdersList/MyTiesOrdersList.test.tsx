import { render } from "@testing-library/react";
import MyTiesOrdersList from ".";

describe("MyTiesOrdersList", () => {
  const orders = [
    {
      _id: "1",
      image: "https://image.jpg",
      date: "2022-02-01",
      price: 10,
      userId: "user",
      sellerId: "seller",
      status: "in progress",
    },
    {
      _id: "2",
      image: "https://image2.jpg",
      date: "2022-02-02",
      price: 20,
      userId: "user",
      sellerId: "seller",
      status: "paid",
    },
  ];

  test("renders correctly", () => {
    const { container } = render(<MyTiesOrdersList orders={orders} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
