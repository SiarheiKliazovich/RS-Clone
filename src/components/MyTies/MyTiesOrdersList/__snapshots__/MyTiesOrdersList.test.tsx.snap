// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTiesOrdersList renders correctly 1`] = `
<div
  class="market-block__products"
>
  <div
    class="row__products"
  >
    <div
      class="tie__product"
    >
      <div
        class="tie__products_img"
      >
        <img
          alt=""
          class="products_img"
          src="https://image.jpg"
        />
      </div>
      <div
        class="tie__products_discription"
      >
        <div
          class="tie__products_name"
        >
          myTiesOrders.date
          : 
          2022-02-01
        </div>
        <div
          class="tie__products_price"
        >
          myTiesOrders.sum
          : 
          10
          $
        </div>
      </div>
    </div>
    <div
      class="tie__product"
    >
      <div
        class="tie__products_img"
      >
        <img
          alt=""
          class="products_img"
          src="https://image2.jpg"
        />
      </div>
      <div
        class="tie__products_discription"
      >
        <div
          class="tie__products_name"
        >
          myTiesOrders.date
          : 
          2022-02-02
        </div>
        <div
          class="tie__products_price"
        >
          myTiesOrders.sum
          : 
          20
          $
        </div>
      </div>
    </div>
  </div>
</div>
`;
